dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/programrule/RuleActionEventMapper.java

```diff
@@ -88,46 +82,43 @@ private List<RuleActionExecutor<Event>> mapRuleEffects(
         .map(
             effect ->
                 buildEventRuleActionExecutor(
-                    effect.ruleId(), effect.data(), effect.ruleAction(), event.getDataValues()))
+                    effect.getRuleId(),
+                    effect.getData(),
+                    effect.getRuleAction(),
+                    event.getDataValues()))
         .filter(Objects::nonNull)
         .filter(
             executor -> isDataElementPartOfProgramStage(executor.getDataElementUid(), programStage))
         .filter(executor -> needsToValidateDataValues(event, programStage))
-        .collect(Collectors.toList());
+        .toList();
   }
 
   private RuleActionExecutor<Event> buildEventRuleActionExecutor(
       String ruleId, String data, RuleAction ruleAction, Set<DataValue> dataValues) {
-    if (ruleAction instanceof RuleActionAssign) {
-      RuleActionAssign action = (RuleActionAssign) ruleAction;
+    if (ruleAction.getType().equals(ProgramRuleActionType.ASSIGN.name())) {
```

Comment:
It was more elegant before but we needed to change this way because `RuleAction` is just one container of information and using types was giving too much responsibility to rule-engine